core/src/main/scala/org/apache/spark/util/JsonProtocol.scala

```diff
@@ -750,15 +751,15 @@ private[spark] object JsonProtocol {
 
   def executorResourceRequestFromJson(json: JValue): ExecutorResourceRequest = {
     val rName = (json \ "Resource Name").extract[String]
-    val amount = (json \ "Amount").extract[Int]
+    val amount = (json \ "Amount").extract[Long]
```

Comment:
`amount` is defined as a long at https://github.com/apache/spark/blob/29258964cae45cea43617ade971fb4ea9fe2902a/core/src/main/scala/org/apache/spark/resource/ExecutorResourceRequest.scala#L52-L58